### ISSUE TITLE
Pytorch classifier

### ISSUE THREAD

**Issue by baderrasheed** [Trusted-AI/adversarial-robustness-toolbox#986]
Is there anyway to load a trained pytorch classifier into PyTorchClassifier?

**Comment by beat-buesser** [comment]
Hi @baderrasheed Thank you very much for using ART!

Yes, you can provide a trained classification `model` with `PyTorchClassifier(model=model)`. This is actually the preferred approach as it allows you to use all the features (e.g. learning rate schedules, multi-GPU, etc.) of a framework like PyTorch to train a model.

**Comment by baderrasheed** [comment]
@beat-buesser  Thank you for your answer, but this does not work. You need to provide the  loss parameter and the model is then uploaded for training and not for predection

**Comment by beat-buesser** [comment]
Hi @baderrasheed That's correct, all the other parameters of `PyTorchClassifier` have to be set accordingly, I have extended my previous post with `...` to indicate this.

The loss has to be provided to ``PyTorchClassifier` also to define the loss that should be attacked.

ART automatically sets the model into '`train` and 'eval` mode according to the operation, e.g. `PyTorchClassifier.predict` automatically sets the model by default into eval mode.

**Comment by baderrasheed** [comment]
I tried but the model was uploaded for training and not for evaluation. Accuracy of the uploaded pretrained model on testset is very low. The loss parameter in the documentation is defined as the loss for training the model and for attack method. Can you please provide an example on how to use this similar to Keras?

**Comment by beat-buesser** [comment]
Hi @baderrasheed Could you please share a code example which reproduces your observations? I would be interested to take a closer look.

**Comment by baderrasheed** [comment]
Hi @beat-buesser My code is a little bit complicated because I am trying to make attacks on the fly but I tried on a simple scenario as well and it didn't work:  Let's say I have a simple pre-trained Pytorch model trained for MNIST classification. I'm trying to upload the model to attack it.  I use PyTorchClassifier(model=model, ...) and then PyTorchClassifier.predict but the accuracy is very low and the weights are not the same for the pretrained model. This works when I try on  a KerasClassifier, but this leads to a lot of problems in other blocks of my code.

**Comment by beat-buesser** [comment]
Hi @baderrasheed With "upload the model to attack it", do you mean transferring it to a GPU?

Would you be able to share an example in a Google Colab notebook? I would be very interested to reproduce it.
